maven-javadoc-plugin/src/main/java/org/apache/maven/plugin/javadoc/AbstractJavadocMojo.java

```diff
@@ -288,6 +289,8 @@
      */
     private static final float SINCE_JAVADOC_1_8 = 1.8f;
 
+    private static final float JAVA_VERSION_FLOAT = JavadocUtil.parseJavadocVersion(SystemUtils.JAVA_VERSION);
```

Comment:
@Tunaki Unfortunately, `JavaVersion` works with floats too. How will this make it any better? Moreover, on needs to extract the version string first before passing to `JavaVersion`.